src/client/SessionPool.cpp

```diff
@@ -48,8 +49,9 @@ ExecutionResponse SessionPool::executeWithParameter(
   if (result.second) {
     auto resp = result.first.executeWithParameter(stmt, parameters);
     if (resp.spaceName != nullptr && *resp.spaceName != config_.spaceName_) {
-      // switch to origin space
-      result.first.execute("USE " + config_.spaceName_);
+      // switch to origin space and try to query again
```

Comment:
What's the error? It's by design, only switch to origin space when space changed.

Reply:
This error will be happened, "Space was not chosen.". 
Just as what I said above, user's query will be failed as only do a query "USE space", but the original stmt is not executed.

Reply:
I can't get your point, please list your operations step by step.

Reply:
1 When a query failed as the connection is broken, we can use retryConnect() to establish a new connection now, but the query will still fail as "Space was not chosen".  if we only retry here with "USE space", the original stmt will fail as it's not run.
2 "if (resp.spaceName != nullptr && *resp.spaceName != config_.spaceName_) ", this statement will only happen when connection is broken and reconnect, as session pool is only inited only with a config space name. space name will not changed after inited, see [here](https://github.com/vesoft-inc/nebula-python/blob/1fe9a44aa7b0f9fdfc35fec12af63e30d7f3546e/README.md?plain=1#L102)
"Each session pool is corresponding to a single USER and a single Space"

Reply:
Why report "Space was not chosen", the `use space` is requested?

Reply:
Yes, it seems almost all the nGQL need "use space" first(only a small number of statements don't, such as "yield 1"), in our case, we use "GO" statement to query.

Reply:
Yes, I means `SessionPool` will switch space automatically.

Reply:
This `if block` is executed when query return `ok` and space changed. So I don't know why it's retried?

Reply:
Maybe you still don't get my idea,

Reply:
I mean with session pool, there is no "space changed" operation

Reply:
I know this.

Reply:
> I mean with session pool, there is no "space changed" operation

There is no such limitation in the program, user can run any query include `use space`. It's just a usage limitation.

Reply:
> This `if block` is executed when query return `ok` and space changed. So I don't know why it's retried?

The query return is not ok, it's an error "Space was not choosen"

Reply:
> > This `if block` is executed when query return `ok` and space changed. So I don't know why it's retried?
> 
> The query return is not ok, it's an error "Space was not choosen"

Only return the error  "Space was not choosen", then, the if block is executed

Reply:
What's your query?

Reply:
> > I mean with session pool, there is no "space changed" operation
> 
> There is no such limitation in the program, user can run any query include `use space`. It's just a usage limitation.
In the link above, there is a description，"If you need to run queries in different spaces, you may have multiple session pools.", so, maybe users will use like that.
Besides, this PR fix a bug as I described before, when connection is broken, even if reconnected(), the original query will fail with the error "Space was not chosen".

Reply:
> What's your query?

My query is "go from xx yield xx" and without using "Use space" every time